modules/eslint-plugin/schematics/ng-add/index.ts

```diff
@@ -4,17 +4,79 @@ import type { Schema } from './schema';
 
 export default function addNgRxESLintPlugin(schema: Schema): Rule {
   return (host: Tree, context: SchematicContext) => {
-    const eslintConfigPath = '.eslintrc.json';
+    const jsonConfigPath = '.eslintrc.json';
+    const flatConfigPath = 'eslint.config.js';
     const docs = 'https://ngrx.io/guide/eslint-plugin';
 
-    const eslint = host.read(eslintConfigPath)?.toString('utf-8');
+    // If a flat config exists, update it
+    if (host.exists(flatConfigPath)) {
+      let content = host.read(flatConfigPath)?.toString('utf-8');
+      if (!content) {
+        context.logger.warn(`Could not read the ESLint flat config at \`${flatConfigPath}\`.`);
+        return host;
+      }
+
+      // Add the import for the NgRx ESLint plugin if not already present.
+      if (!content.includes("from '@ngrx/eslint-plugin'")) {
+        content = `import ngrxEslintPlugin from '@ngrx/eslint-plugin';\n` + content;
+      }
+
+      // Create the NgRx configuration to be added.
+      // (This mirrors your current JSON override but adapted to flat config syntax.)
+      const ngrxConfig = `
+{
+  files: ['*.ts'],
+  extends: [\`plugin:@ngrx/${schema.config}\`],
+},`;
+
+      // Check if the flat config already contains an NgRx config.
+      if (!content.includes('plugin:@ngrx')) {
+        // Look for an exported default array.
+        const exportDefaultRegex = /export\s+default\s+\[([\s\S]*?)\];/;
```

Comment:
Same comment here, ideally we also support `module.exports = ...`.
As I expect most eslint configs use tseslint, checking for `tseslint.config` is probably the simplest solution. Updating it is harder than a JSON file, I think it can be easier if we use the TS AST to find the `tseslint.config` CallExpression, and add an argument.